blockly-core/core/ui/contract_editor/type_dropdown.js

```diff
@@ -76,12 +85,22 @@ Blockly.TypeDropdown.prototype.createSelect_ = function() {
   this.typeChoices_.forEach(function (choiceKey) {
     var menuItem = new goog.ui.MenuItem(choiceKey);
     newTypeDropdown.addItem(menuItem);
-    this.setMenuItemColor_(menuItem, Blockly.FunctionalTypeColors[choiceKey]);
+    var colorIconDiv = this.createColorSquareDiv();
+    goog.dom.insertChildAt(menuItem.getElement(), colorIconDiv, 0);
+    this.setSquareIconColor(choiceKey, colorIconDiv);
   }, this);
   newTypeDropdown.setValue(this.type_);
   return newTypeDropdown;
 };
 
+/**
+ * @returns {Element}
+ * @private
+ */
+Blockly.TypeDropdown.prototype.createColorSquareDiv = function () {
+  return goog.dom.createDom('div', "color-square-icon");
```

Comment:
Doesn't really matter, but feels strange switching between single and double quotes on the same line here.